packages/core/logger/src/index.js

```diff
@@ -33,137 +31,98 @@ export class Logger {
+    this.logFunction(...Array.from(args), 'red');
+    this.writeToFile("[error]: ", args);
   }
-  this.events.emit("log", "error", ...arguments);
-  this.logFunction(...Array.from(arguments), 'red');
-  this.writeToFile("[error]: ", ...arguments);
-};
 
-Logger.prototype.warn = function () {
-  if (!arguments.length || !(this.shouldLog('warn'))) {
-    return;
+  warn(...args) {
+    if (!args.length || !(this.shouldLog('warn'))) {
+      return;
+    }
+    this.events.emit("log", "warn", args);
+    this.logFunction(...Array.from(args), 'yellow');
+    this.writeToFile("[warning]: ", args);
   }
-  this.events.emit("log", "warn", ...arguments);
-  this.logFunction(...Array.from(arguments), 'yellow');
-  this.writeToFile("[warning]: ", ...arguments);
-};
 
-Logger.prototype.info = function () {
-  if (!arguments.length || !(this.shouldLog('info'))) {
-    return;
+  info(...args) {
+    if (!args.length || !(this.shouldLog('info'))) {
+      return;
+    }
+    this.events.emit("log", "info", args);
+    this.logFunction(...Array.from(args), 'green');
+    this.writeToFile("[info]: ", args);
   }
-  this.events.emit("log", "info", ...arguments);
-  this.logFunction(...Array.from(arguments), 'green');
-  this.writeToFile("[info]: ", ...arguments);
-};
 
-Logger.prototype.consoleOnly = function () {
-  if (!arguments.length || !(this.shouldLog('info'))) {
-    return;
+  consoleOnly(...args) {
+    if (!args.length || !(this.shouldLog('info'))) {
+      return;
+    }
+    this.logFunction(...Array.from(args), 'green');
+    this.writeToFile("[consoleOnly]: ", args);
   }
-  this.logFunction(...Array.from(arguments), 'green');
-  this.writeToFile("[consoleOnly]: ", ...arguments);
-};
 
-Logger.prototype.debug = function () {
-  if (!arguments.length || !(this.shouldLog('debug'))) {
-    return;
+  debug(...args) {
+    if (!args.length || !(this.shouldLog('debug'))) {
+      return;
+    }
+    this.events.emit("log", "debug", args);
+    this.logFunction(args, null);
+    this.writeToFile("[debug]: ", args);
   }
-  this.events.emit("log", "debug", ...arguments);
-  this.logFunction(...arguments, null);
-  this.writeToFile("[debug]: ", ...arguments);
-};
 
-Logger.prototype.trace = function () {
-  if (!arguments.length || !(this.shouldLog('trace'))) {
-    return;
+  trace(...args) {
+    if (!args.length || !(this.shouldLog('trace'))) {
+      return;
+    }
+    this.events.emit("log", "trace", args);
+    this.logFunction(args, null);
+    this.writeToFile("[trace]: ", args);
   }
-  this.events.emit("log", "trace", ...arguments);
-  this.logFunction(...arguments, null);
-  this.writeToFile("[trace]: ", ...arguments);
-};
 
-Logger.prototype.dir = function (txt) {
-  if (!txt || !(this.shouldLog('info'))) {
-    return;
+  dir(txt) {
+    if (!txt || !(this.shouldLog('info'))) {
+      return;
+    }
+    this.events.emit("log", "dir", txt);
+    this.logFunction(txt, null);
+    this.writeToFile("[dir]: ", ...arguments);
```

Comment:
In some calls the spread (`...`) operator is used on the arguments to `writeToFile` but in other calls the array is being passed. Should they all be consistent?

Reply:
Yes indeed. Updating.